src/main/java/de/danielbechler/diff/access/PropertyAwareAccessor.java

```diff
@@ -29,4 +29,8 @@
 	Set<Annotation> getReadMethodAnnotations();
 
 	<T extends Annotation> T getReadMethodAnnotation(Class<T> annotationClass);
+
+   <T extends Annotation> T getAnnotation(Class<T> annotationClass);
+
+   int getFieldModifiers();
```

Comment:
I think this method bleeds to much low-level information to the outside world. It doesn't really serve as an abstraction on top of the core JDK method, is hard to understand without reading documentation and could cause compatibility problems between different JDKs. It would be better to explicitly provide methods like `isTransient`, etc. But I don't like that either, because it wouldn't be applicable to properties that only have a getter but no corresponding field (e.g. it's private, has a different name, etc.).

How about a different approach: a config setting that allows to exclude transient properties form introspection. This way the exclusion can be implemented deep down in the `Introspector`, which is nice since it's supposed to be the only place to have unlimited access to the low-level class system information anyway.

Would that work for you? Do you need other modifiers?